src/librustc_resolve/resolve_imports.rs

```diff
@@ -767,12 +777,31 @@ impl<'a, 'b:'a, 'c: 'b> ImportResolver<'a, 'b, 'c> {
                 if import.is_uniform_paths_canary {
                     continue;
                 }
-
                 let error = ResolutionError::UnresolvedImport(None);
                 resolve_error(self.resolver, import.span, error);
                 break;
             }
+        } else if !error_vec.is_empty() {
+            self.throw_unresolved_import_error(error_vec);
+        }
+    }
+
+    fn throw_unresolved_import_error(&self, error_vec: Vec<(Span, String, String)>) {
+        let multi_span = MultiSpan::from_spans(error_vec.clone().into_iter()
+                                    .map(|elem: (Span, String, String)| { elem.0 }
+                                    ).collect());
+        let path_vec: Vec<String> = error_vec.clone().into_iter()
+                        .map(|elem: (Span, String, String)| { format!("`{}`", elem.1) }
+                        ).collect();
+        let path = path_vec.join(", ");
+        let msg = format!("unresolved import{} {}",
+                        if path_vec.len() > 1 { "s" } else { "" },  path);
+        let mut err = struct_span_err!(self.resolver.session,
+                                       multi_span.clone(), E0432, "{}", &msg);
```

Comment:
To avoid the error you're seeing in travis you need to extract this to a method called from both here and the other occurrence of E0432.

Reply:
@estebank Sure!

Reply:
@estebank Challenge here is to create a generic function which can take any one of `Span` or `MultiSpan` as a type of argument. I think creating a `traits` and implementing it both for `Span` and `MultiSpan` is one way but I am bit reluctant toward this approach. Let me know what are your thoughts on that.

Reply:
done!